virtual/vbox/vbox.go

```diff
@@ -146,6 +146,23 @@ func (vm *vm) Stop() error {
 	return nil
 }
 
+// Will call savestate on vm
+func (vm *vm) Suspend(ctx context.Context) error {
+	_, err := VBoxCmdContext(ctx, vboxCtrlVM, vm.id, "savestate")
+	if err != nil {
+		log.Error().
+			Str("ID", vm.id).
+			Msgf("Failed to suspend VM: %s", err)
+		return err
+	}
+
+	log.Debug().
+		Str("ID", vm.id).
+		Msgf("Suspended vm")
+
+	return nil
+}
```

Comment:
There is no problem on saving the state of VM however have you check that it becomes available when you try to wake it up ?

Reply:
On host, rather than test cases, it  acts differently.

Reply:
Pretty sure i tested i on my own computer, and that it worked fine.
But will test it again just to be sure.

Reply:
I can provide you an access to test environment, if you would like to test it on server instead of your local, send me your public key.